### PR TITLE
Add 3 Scopes, Devise and fix engine

### PR DESCRIPTION
**Pull request by gwmoura**
* Fix engine removing `lib/railsblocks/auth.rb` and adding `lib/railsblocks-auth.rb`
* Add gem 'devise'
* Add scopes:
  * User
  * Admin
  * Manager
* Add one test for each scope

Commit 0836f58
* Add factories for User, Admin and Manager
* Add mixin for all scopes

Commit 1d04764
* Remove registration porperty in model for Admin and Manager
* Add tests for requests
* Add tests for excluded routes (scopes Admin and Manager)
* Add json as default format in test env - `spec/internal/config/routes.rb`

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -242,6 +256,3 @@ DEPENDENCIES
   shoulda-matchers (~> 2.8)
   sqlite3 (~> 1.3)
   thor (~> 0.19)
-
```
Comment (fernandes) [inline Gemfile.lock]: why removed? upgrade your bundler to 1.10.6

File: app/models/manager.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    if authentication_token.blank?
+      self.authentication_token = generate_authentication_token
+    end
+  end
+
+  private
+    def generate_authentication_token
+      loop do
+        token = Devise.friendly_token
+        break token unless User.where(authentication_token: token).first
```
Comment (fernandes) [inline app/models/manager.rb]: `User.where` here?

File: app/models/admin.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    if authentication_token.blank?
+      self.authentication_token = generate_authentication_token
+    end
+  end
+
+  private
+    def generate_authentication_token
+      loop do
+        token = Devise.friendly_token
+        break token unless User.where(authentication_token: token).first
```
Comment (fernandes) [inline app/models/admin.rb]: `User.where` here?

File: spec/factories/railsblocks_admins.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :railsblocks_admin, :class => 'Railsblocks::Admin' do
```
Comment (fernandes) [inline spec/factories/railsblocks_admins.rb]: create some factory please

File: spec/factories/railsblocks_managers.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :railsblocks_manager, :class => 'Railsblocks::Manager' do
```
Comment (fernandes) [inline spec/factories/railsblocks_managers.rb]: create some factory please

File: spec/factories/railsblocks_users.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :railsblocks_user, :class => 'Railsblocks::User' do
```
Comment (fernandes) [inline spec/factories/railsblocks_users.rb]: create some factory please

**Comment by fernandes** [comment]
we *must* not allow manager and admin sign up...

File: Gemfile.lock (hunk 1 above)
Comment (gwmoura) [inline Gemfile.lock]: :+1:

File: app/models/manager.rb (hunk 2 above)
Comment (gwmoura) [inline app/models/manager.rb]: :+1:

File: app/models/admin.rb (hunk 3 above)
Comment (gwmoura) [inline app/models/admin.rb]: :+1:

File: spec/factories/railsblocks_admins.rb (hunk 4 above)
Comment (gwmoura) [inline spec/factories/railsblocks_admins.rb]: :+1:

File: spec/factories/railsblocks_managers.rb (hunk 5 above)
Comment (gwmoura) [inline spec/factories/railsblocks_managers.rb]: :+1:

File: spec/factories/railsblocks_users.rb (hunk 6 above)
Comment (gwmoura) [inline spec/factories/railsblocks_users.rb]: :+1:

File: app/controllers/railsblocks/auth/admin_controller.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+module Railsblocks::Auth
+  class AdminController < ApplicationController
+    include UserAuth
```
Comment (fernandes) [inline app/controllers/railsblocks/auth/admin_controller.rb]: if you wanna include something on a __controller__, add inside `controllers` folder, or `lib/controllers`

**Pull request merged**